module/zfs/dsl_pool.c

```diff
@@ -555,6 +563,15 @@ dsl_pool_create(spa_t *spa, nvlist_t *zplprops __attribute__((unused)),
 #endif
 	dsl_dataset_rele_flags(ds, DS_HOLD_FLAG_DECRYPT, FTAG);
 
+	if (spa_is_shared_log(spa)) {
+		dp->dp_chain_map_obj = zap_create_flags(dp->dp_meta_objset, 0,
+		    ZAP_FLAG_HASH64 | ZAP_FLAG_UINT64_KEY,
```

Comment:
probably doesn't matter much, but since the key is the random spa_guid, I think you can also specify `ZAP_FLAG_PRE_HASHED_KEY`